src/python/pants/backend/go/goals/package_binary_integration_test.py

```diff
@@ -106,6 +112,198 @@ def test_package_simple(rule_runner: RuleRunner) -> None:
     assert result.stdout == b"Hello world!\n"
 
 
+def _gen_third_party_pkg():
+    # Implements hashing algorithm from https://cs.opensource.google/go/x/mod/+/refs/tags/v0.5.0:sumdb/dirhash/hash.go.
+    def _compute_module_hash(files: Iterable[tuple[str, str]]) -> str:
```

Comment:
This function shows up in multiple places. Maybe just put it in one place so the various call sites can share?

Reply:
For that matter, maybe have a single function which can take the files to put in the packages, and then computes the zip file bytes, go.sum entry, etc. returned in a probably a single dataclass.

Reply:
👍 I was considering it but felt like maybe it was too few uses.

Reply:
It's only a few uses, but the hash algorithm is pretty particular in its details, so having it defined once will be good for future readers of the code (and for preventing any bugs due to copying mistakes).

Reply:
The function could even return a dictionary of name to file bytes which can then be passed to `rule_runner.write_files`.

Reply:
Done!